# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.cpp
# Commit: d46f0bd36d7e
# Full Hash: d46f0bd36d7edd1d9d040be6298e6e5d1ccc70b2
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-01-05 09:33:53
# Regressor Bug: 1737944
# File Overlap Count: 1
# Description:
#   Bug 1737944 - P9: Make AccGroupInfo use Accessible. r=Jamie
#   
#   Now that GroupPosition and GetGroupInfo are in Accessible, we can make
#   AccGroupInfo use the Accessible base class.
#   
# ==============================================================================

diff -r ac04b727a98b -r d46f0bd36d7e accessible/generic/LocalAccessible.cpp
--- a/accessible/generic/LocalAccessible.cpp	Tue Jan 04 21:01:37 2022 +0000
+++ b/accessible/generic/LocalAccessible.cpp	Tue Jan 04 21:01:38 2022 +0000
@@ -1978,11 +1978,12 @@
       if (roleMapEntry && (roleMapEntry->role == roles::OUTLINEITEM ||
                            roleMapEntry->role == roles::LISTITEM ||
                            roleMapEntry->role == roles::ROW)) {
-        LocalAccessible* parent = const_cast<LocalAccessible*>(this)
-                                      ->GetOrCreateGroupInfo()
-                                      ->ConceptualParent();
+        Accessible* parent = const_cast<LocalAccessible*>(this)
+                                 ->GetOrCreateGroupInfo()
+                                 ->ConceptualParent();
         if (parent) {
-          rel.AppendTarget(parent);
+          MOZ_ASSERT(parent->IsLocal());
+          rel.AppendTarget(parent->AsLocal());
         }
       }
 
